kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -200,7 +200,8 @@ class BatchJobSubmission(
       val process = builder.start
       applicationInfo = currentApplicationInfo
       while (!applicationFailed(applicationInfo) && process.isAlive) {
-        if (!appStatusFirstUpdated && applicationInfo.isDefined) {
+        if (!appStatusFirstUpdated && applicationInfo.isDefined &&
+          !applicationInfo.exists(_.state == ApplicationState.NOT_FOUND)) {
```

Comment:
is this expected ? we can not find application after submission in a short time.

Reply:
if it is a corner case, I think we can add a timeout and mark it failed if application still not found.